Feature Flag Image Uploads

ardelato opened the pull request:
## Description

We will be disabling the ability to upload images for the moment for two reasons:
1. The functionality is still buggy - https://github.com/iFixit/restarters/pull/10#issuecomment-2847910779
2. We are still looking into where else to store image uploads in ArgoCD - https://ifixit.slack.com/archives/C05SC2972JZ/p1746651439515929?thread_ts=1746648008.262179&cid=C05SC2972JZ

The bigger blocker is reason 2. Even if we were to fix image uploading, we would be potentially switching from using a `PVC` mounted to `/storage` to something external. In which case, images that would have been uploaded, might end up being wiped.

qa_req 0

ardelato committed:
fix(controllers): update FixometerFile import paths in controllers

Fixed the import statements for FixometerFile in multiple controllers to
use the correct namespace.

ardelato committed:
feat(image-upload): conditionally load images for events devices

ardelato committed:
feat(env): add FEATURE__IMAGE_UPLOAD environment variable for image upload control

Introduced a new environment variable FEATURE__IMAGE_UPLOAD to manage the
image upload feature.

ardelato committed:
feat(image-uploads): disable image upload components in blade templates

ardelato committed:
feat(image-uploads): disable image upload components in Vue components

djmetzle reviewed (approved):
This seems reasonable @ardelato.

To ask, the language in PR description here seems confused. We're looking to using an external filesystem, as the pod local FS isnt shared, and is practically ephemeral. Our standard practice is to use [S3](https://laravel.com/docs/12.x/filesystem#amazon-s3-compatible-filesystems). The file pointer object in the DB seems irrelevant. 

Otherwise, this seems good CR :flags:

ardelato committed:
fix(image-uploads): use global var to retrieve image upload enabled for Vue components

If we build the Vue components then we can't utilize the
PHP if statements in Vue components. In dev we can because we utilize
a dev server.

As such, we need to do something similar like with the window.appInstance
and window.appDebug to retrieve the image upload enabled variable in
direct components. Otherwise, we'd need to propogate the variable, as a
prop, across multiple components.

Following the 3's rule, we'll create a parent global variable named
window.Laravel that will, in the future, hold all the Laravel specific
variables that we need to access in the Vue components. For now, we will
just store the image upload enabled variable and in another pull request
we'll add the other variables.

djmetzle commented on resources/views/layouts/header_plain.blade.php:
```diff
@@ -47,6 +47,9 @@
         <script>
             window.appInstance = "{{ env('APP_INSTANCE', 'base') }}";
             window.appDebug = "{{ env('APP_DEBUG', '0') }}";
+            window.Laravel = {
```
This is intended as a single monolithic global variable, intended for frontend config? In the spirit of our monorepo JS `App` variable?

window.App or window.Global or window.Framework seems like a code smell. We can _at least_ segment by some namespace or feature, for modularity.

I recommend we avoid this approach. But that's not a blocker either.

ardelato replied:
This would mainly be for the frontend config.

I agree this is not the best approach. We could probably use Vue's [Provide/Inject]( https://vuejs.org/guide/components/provide-inject) feature instead.

That said, I'm still getting familiar with the usage and setup of Vue with this Laravel app, so this might just be an interim solution that cleans up the current globals pattern.

djmetzle reviewed (approved):
CR :flags:

mlahargou commented on resources/views/layouts/header.blade.php:
```diff
@@ -54,6 +54,9 @@
       window.appInstance = "{{ env('APP_INSTANCE', 'base') }}";
       window.appDebug = "{{ env('APP_DEBUG', '0') }}";
+      window.Laravel = {
+        imageUploadEnabled: @json(config('restarters.features.image_upload_enabled', false)),
```
Is there a reason we default to false? I would think true, as to keep current behavior when we merge this upstream.

ardelato replied:
This probably won't be merged into the upstream anytime soon because this was related to our Laravel upgrades 10-12 as well as our deployment process of using K8S and ArgoCD. 

Once we figure our where we want to store the image uploads in K8s and confirm image uploading actually works, will we turn this back on. At which point, we can set the default back to true.

djmetzle replied:
> [..] where we want to store the image uploads ~~in~~ **outside of** K8s and [..]

FTFY...

mlahargou reviewed (approved):
CR 📱
